src/analysis/retail/evoker/devastation/modules/abilities/ShatteringStar.tsx

```diff
@@ -0,0 +1,125 @@
+import Events, { DamageEvent } from 'parser/core/Events';
+
+const {
+  DISINTEGRATE,
+  FIRE_BREATH,
+  FIRE_BREATH_FONT,
+  ETERNITY_SURGE,
+  ETERNITY_SURGE_FONT,
+  LIVING_FLAME_DAMAGE,
+  LIVING_FLAME_CAST,
+  AZURE_STRIKE,
+  SHATTERING_STAR,
+} = SPELLS;
+
+const { DRAGONRAGE_TALENT } = TALENTS_EVOKER;
+
+type DamageCounter = {
+  disintegrate: number;
+  fireBreath: number;
+  eternitySurge: number;
+  livingFlame: number;
+  azureStrike: number;
+};
+
+class ShatteringStar extends Analyzer {
+  totalBreaths: number = 0;
+  totalApplications: number = 0;
+  totalCasts: number = 0;
+  isBuffOn: boolean = false;
+  inDragonRageWindow: boolean = false;
+  damageCounters: {
+    [window: number]: DamageCounter;
+  } = {};
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(
+      Events.applydebuff.by(SELECTED_PLAYER).spell(SHATTERING_STAR),
+      this.onApply,
+    );
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(DRAGONRAGE_TALENT), () => {
+      this.inDragonRageWindow = true;
+    });
+
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(DRAGONRAGE_TALENT), () => {
+      this.inDragonRageWindow = false;
+    });
+
+    this.addEventListener(
+      Events.removedebuff.by(SELECTED_PLAYER).spell([SHATTERING_STAR]),
+      this.onRemove,
+    );
+
+    this.addEventListener(
+      Events.damage
+        .by(SELECTED_PLAYER)
+        .spell(
+          [
+            DISINTEGRATE,
+            FIRE_BREATH,
+            FIRE_BREATH_FONT,
+            ETERNITY_SURGE,
+            ETERNITY_SURGE_FONT,
+            LIVING_FLAME_DAMAGE,
+            LIVING_FLAME_CAST,
+            AZURE_STRIKE,
+          ].map((spell) => ({ id: spell.id })),
+        ),
+      this.onDamage,
+    );
+  }
+
+  onApply() {
+    if (this.inDragonRageWindow) {
+      return;
+    }
+
+    this.totalCasts += 1;
+    this.isBuffOn = true;
+    this.damageCounters[this.totalCasts] = {
+      disintegrate: 0,
+      fireBreath: 0,
+      eternitySurge: 0,
+      livingFlame: 0,
+      azureStrike: 0,
+    };
+  }
+
+  onRemove() {
+    this.isBuffOn = false;
+  }
+
+  onDamage(damageEvent: DamageEvent) {
+    if (!this.isBuffOn) {
+      return;
+    }
+
+    const currentWindow = this.damageCounters[this.totalCasts];
+
+    switch (damageEvent.ability.name) {
+      case DISINTEGRATE.name:
+        currentWindow.disintegrate += damageEvent.amount;
+        break;
+      case FIRE_BREATH.name:
+        currentWindow.fireBreath += damageEvent.amount;
+        break;
+      case ETERNITY_SURGE.name:
+        currentWindow.eternitySurge += damageEvent.amount;
+        break;
+      case LIVING_FLAME_DAMAGE.name:
+        currentWindow.fireBreath += damageEvent.amount;
+        break;
+      case AZURE_STRIKE.name:
+        currentWindow.fireBreath += damageEvent.amount;
+        break;
+      default:
+        break;
+    }
```

Comment:
We usually prefer to differentiate spells by guid, not name. Is there a particular reason you're using name here?

Reply:
I initially did this because depending on the talent, the empowered spell ID's will change, but then I realized I can just add another case 🤦

Reply:
Oh that's interesting, I had no idea charging up changed the spellID for Evoker. I'd still prefer it's done with guid - if you find yourself often grouping the same IDs, you can even add a const to a shared file, the way many specs do with `constants.ts`. Up to you how you want to style it though.

Reply:
name is also susceptible to internationalization issues. I don't remember if we force translation on WoWA, but if we don't then on non-english logs you may get non-english names.